lib/CodeGen/CGCall.cpp

```diff
@@ -4113,8 +4118,14 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
     if (IRFunctionArgs.hasInallocaArg() &&
         i == IRFunctionArgs.getInallocaArgNo())
       continue;
-    if (i < IRFuncTy->getNumParams())
-      assert(IRCallArgs[i]->getType() == IRFuncTy->getParamType(i));
+    if (i < IRFuncTy->getNumParams()) {
+      if (IRCallArgs[i]->getType() != IRFuncTy->getParamType(i)) {
+        llvm::errs() << *CalleePtr << " arg" << i << ": "
```

Comment:
This debug message is only output in debug build when there is a mismatch between argument and parameter and an assertion will happen immediately. Since in this situation the program will crash any way, a more detailed debug message should not be a problem. In release build or in debug build but without the issue, there will be no extra output.

I found this debug output helped a lot for debugging such issues. Since HCC is still not stable enough, I expect we may need this debug message for some time. After HCC is stable we may remove this.